src/test/java/com/exasol/bucketfs/SyncAwareBucketIT.java

```diff
@@ -64,62 +43,80 @@ private Path createTestFile(final Path tempDir, final String fileName, final int
     }
 
     // [itest->dsn~uploading-to-bucket~1]
-    @ValueSource(strings = { "dir1/", "dir2/sub2/", "dir3/sub3/subsub3/", "/dir4/", "/dir5/sub5/" })
     @ParameterizedTest
-    void testUploadToDirectoryInBucket(final String pathInBucket, @TempDir final Path tempDir)
-            throws BucketAccessException, InterruptedException, IOException, TimeoutException {
+    @ValueSource(strings = { "dir1/", "dir2/sub2/", "dir3/sub3/subsub3/", "/dir4/", "/dir5/sub5/" })
+    void testUploadToDirectoryInBucket(final String pathInBucket, @TempDir final Path tempDir) throws Exception {
         final var fileName = "file.txt";
         final var file = createTestFile(tempDir, fileName, 1);
-        final var bucket = createDefaultBucket();
+        final var bucket = getDefaultBucketForWriting();
         bucket.uploadFile(file, pathInBucket);
-        assertThat(createDefaultBucket().listContents(pathInBucket), Matchers.contains(fileName));
+        assertThat(getDefaultBucketForWriting().listContents(pathInBucket), Matchers.contains(fileName));
```

Comment:
What this does not test is what happens if the path contains more than one file. That listing could still be wrong.

Reply:
See [doc](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#contains(E...)): 
> For a positive match, the examined iterable must be of the same length as the number of specified items.